flink-runtime/src/main/java/org/apache/flink/runtime/util/ClusterUncaughtExceptionHandler.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.util;
+
+import org.apache.flink.configuration.ClusterOptions;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility for handling any uncaught exceptions
+ *
+ * <p>Handles any uncaught exceptions according to cluster configuration in {@link ClusterOptions}
+ * to either just log exception, or fail job.
+ */
+public class ClusterUncaughtExceptionHandler implements Thread.UncaughtExceptionHandler {
```

Comment:
I think it would be good to add a test for the exit behavior using the `SystemExitTrackingSecurityManager`.

Reply:
Hi @tillrohrmann, I tried to add a test to do using the `SystemExitTrackingSecurityManager`, but the problem is that in `FAIL` mode, the `ClusterUncaughtExceptionHandler `invokes the `FatalExitExceptionHandler`, which will in turn invoke `FlinkSecurityManager.forceProcessExit(-17)`, which in turn will invoke `setSecurityManager(null)`, thus making it difficult to test exit behaviour this way. 
I have just added an integration test instead using the `TestJvmProcess` instead, to check exit behaviour. Do check that and let me know if there is a better way to test this 👍

Reply:
Sounds good. I didn't know about the specific behaviour of the `FatalExitExceptionHandler`. Thanks a lot for sharing this knowledge.